update items.validtion to allow single entry inserts to existing array

eric-foster-angi opened the pull request:
## Summary

Implements fix for issue: https://github.com/contentful/contentful-migration/issues/1166

When updating the valid Content Types for a reference Content Model field, you cannot just insert or add a single entry as `validations` will completely write over the existing entries.  This requires copying and pasting the list which introduces the risk that someone might do it wrong and wipe out needed values.

Instead, we should be able to just add/insert a new value to the array, reducing risk and headache.

## Description

implements a new method **addItemsValidation** that allows you to update existing validations on items without completely writing over the existing values.

## Motivation and Context
Given an existing model defined as:
```typescript
  testModel
    .createField('relatedEntries')
    .name('Related Entries')
    .type('Array')
    .items({
      type: 'Link',
      linkType: 'Entry',
      validations: [{ linkContentType: ['contentModelA', 'contentModelB']}]
    })
```

Today if we wanted to add contentModelC to this list of validations, we must redefine the entire array:
```typescript
  testModel
    .editField('relatedEntries')
    .items({
      type: 'Link',
      linkType: 'Entry',
      validations: [{ linkContentType: ['contentModelA', 'contentModelB', 'contentModelC']}]
    })

```

Instead it would be desirable to be able to add a single entry:
```typescript
  const testModel = migration.editContentType('testModel')
  testModel
    .editField('relatedEntries')
    .addItemsValidation([{ linkContentType: ['contentModelC'] }])
```


## Todos
None

## Screenshots 
<img width="882" alt="skills-validation-atleast-1" src="https://github.com/user-attachments/assets/afda80f7-3d16-4e7c-aa5b-1e370df32b70" />
<img width="925" alt="related-entries-validations-after-update" src="https://github.com/user-attachments/assets/bf8b2b1a-5f14-4962-a5f1-e68161178959" />
<img width="977" alt="test-model-fields" src="https://github.com/user-attachments/assets/a3a41c08-c674-404f-9760-c34b3341e6be" />
<img width="1328" alt="new-content-models-created" src="https://github.com/user-attachments/assets/b58d8d5d-85a5-495d-bd17-851f12a1d100" />
(if appropriate):

ethan-ozelius-contentful commented:
Looks like there might be some failing tests?

```
 // npm run test:unit
  337 passing (538ms)
  1 pending
  3 failing

  1) FieldAddItemsValidationIntent
       when adding validations to array field items
         adds validations to existing items validations:
     TypeError: callsite.getFileName is not a function
      at Object.addItemsValidation (src/lib/migration-steps/action-creators.ts:739:29)
      at Context.<anonymous> (test/unit/lib/intent/field-add-items-validation.spec.ts:12:74)
      at processImmediate (node:internal/timers:476:21)

  2) FieldAddItemsValidationIntent
       when adding validations to array field items
         creates items validations array if it does not exist:
     TypeError: callsite.getFileName is not a function
      at Object.addItemsValidation (src/lib/migration-steps/action-creators.ts:739:29)
      at Context.<anonymous> (test/unit/lib/intent/field-add-items-validation.spec.ts:51:74)
      at processImmediate (node:internal/timers:476:21)

  3) FieldAddItemsValidationIntent
       when adding validations to array field items
         creates items object if it does not exist:
     TypeError: callsite.getFileName is not a function
      at Object.addItemsValidation (src/lib/migration-steps/action-creators.ts:739:29)
      at Context.<anonymous> (test/unit/lib/intent/field-add-items-validation.spec.ts:88:74)
      at processImmediate (node:internal/timers:476:21)

Exited with code exit status 3
```

```
// npm run lint


> contentful-migration@0.0.0-development lint
> eslint 'examples/**/*.js' 'test/**/*.js' 'src/**/*.js' && tslint --project tsconfig-dev.json --config tslint.json -e '**/*.js'

strict-type-predicates does not work without --strictNullChecks
/home/circleci/project/test/unit/lib/actions/field-add-items-validation.spec.ts:89:7
ERROR: 89:7   no-unused-expression  unused expression, expected an assignment or function call

/home/circleci/project/test/unit/lib/intent/field-add-items-validation.spec.ts:117:7
ERROR: 117:7  no-unused-expression  unused expression, expected an assignment or function call
```

```
// npm run test:integration
  47 passing (57s)
  1 failing

  1) the migration
       updates items validation for existing field:
     AssertionError: expected [ { linkContentType: [ …(3) ] } ] to deep include { linkContentType: [ 'contentModelC' ] }
      at Context.<anonymous> (test/integration/migration.spec.js:1466:59)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

ethan-ozelius-contentful commented on src/lib/action/field-add-items-validation.ts:
```diff
@@ -0,0 +1,60 @@
+    const fields = ct.fields
+    const field = fields.getField(this.getFieldId())
+
+    if (!field.items) {
```
field would only have a `items` property if it is of type `reference` or `media`.  So we might want to add one more check here to make sure we don't unnecessarily add the items field to a field where it wouldn't belong.  Something like:

```js
if (!field.items && ['Array', 'Link'].includes(field.type)) {
```

eric-foster-angi commented:
> > Sorry for the delay, rebased with master, fixed tests, and updated based on feedback. How can I run the tests here for the MR?
> 
> Hey Eric Foster (@eric-foster-angi) we can help you run the PR checks and get this merged in after the holidays. We will sync back with you the week of 12/29.

Any idea where this landed?
